app/controllers/types_controller.rb

```diff
@@ -0,0 +1,63 @@
+class TypesController < ApplicationController
+  before_action :authenticate_user!
+  load_and_authorize_resource
+  before_action :set_type, only: [:show, :edit, :update, :destroy]
+  respond_to :js, :html
+
+  def index
+    @types = current_user.types.order('created_at DESC').page(params[:page]).per(4)
+
+    @types = if params[:show_all] == 'true'
+               Type.all.order('created_at DESC').page(params[:page]).per(4)
+             else
+               current_user.types.order('created_at DESC').page(params[:page]).per(4)
+             end
+  end
+
+  def show
+  end
+
+  def new
+    @type = Type.new
+    @type.user_id = current_user.id
+  end
+
+  def edit
+  end
+
+  def create
+    @type = Type.new(type_params)
+    @type.user_id = current_user.id
+    render :new if @type.save != true
+  end
+
+  def update
+    render :edit if @type.update(type_params_update) != true
```

Comment:
@VanjaJeftic consider using this syntax:

```suggestion
    render :edit unless @type.update(type_params_update)
    # or
    saved = @type.update(type_params_update)
    render :edit unless saved
```